ai_ref_kits/agentic_multimodal_travelplaner/start_mcp_servers.py

```diff
@@ -0,0 +1,435 @@
+"""MCP server startup script for the agentic multimodal travel planner.
+
+This module manages MCP (Model Context Protocol) servers, providing
+functionality to start, stop, and download server scripts from GitHub.
+"""
+
+import socket
+import subprocess
+import sys
+import time
+import urllib.error
+import urllib.request
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import yaml
+
+CONFIG_PATH = Path("config/mcp_config.yaml")
+LOG_DIR = Path("logs")
+LOG_DIR.mkdir(exist_ok=True)
+
+# GitHub URLs for known MCP servers
+GITHUB_MCP_URLS = {
+    "ai_builder_mcp_flights.py": (
+        "https://raw.githubusercontent.com/intel/intel-ai-assistant-builder/"
+        "main/mcp/mcp_servers/mcp_google_flight/server.py"
+    ),
+    "ai_builder_mcp_hotel_finder.py": (
+        "https://raw.githubusercontent.com/intel/intel-ai-assistant-builder/"
+        "main/mcp/mcp_servers/mcp_google_hotel/server.py"
+    ),
+}
+
+
+def is_port_in_use(port: int) -> bool:
+    """Check if a port is currently in use.
+
+    Args:
+        port: The port number to check.
+
+    Returns:
+        True if the port is in use, False otherwise.
+    """
+    try:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.settimeout(0.5)
+            return sock.connect_ex(("127.0.0.1", port)) == 0
+    except OSError:
+        return False
+
+
+def kill_processes_on_port(port: int) -> None:
+    """Kill any processes using the specified port.
+
+    Args:
+        port: The port number to clear.
+    """
+    try:
+        result = subprocess.run(
+            ["lsof", "-t", f"-i:{port}"],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        if result.stdout.strip():
+            pids = result.stdout.strip().split("\n")
+            for pid in pids:
+                if pid:
+                    subprocess.run(["kill", "-9", pid], check=False)
+                    print(f"Killed process {pid} on port {port}")
+    except FileNotFoundError:
+        # lsof not available; best effort skip
+        pass
```

Comment:
The code seems to be the same as in start_agents. If it's the case, move it to the separate file (utils?) and use it in both scripts.